test/widgets/profile_test.dart

```diff
@@ -365,4 +377,270 @@ void main() {
+      prepareRequestSuccess(Duration(milliseconds: 100));
+      scheduleEventAfter(Duration(seconds: 10), true);
+      await doToggle(tester, toggleInteractionMode);
+      await tester.pump();
+      await tester.pump();
+      checkAppearsActive(tester, true);
+      checkRequest(true);
+
+      // Local-echo timeout passes and event hasn't come; change back.
+      await tester.pump(RemoteSettingBuilder.localEchoIdleTimeout);
+      await tester.pump();
+      checkAppearsActive(tester, false);
+
+      // The event comes after a while; update for the new value.
+      await tester.pump(Duration(seconds: 10));
+      check(connection.takeRequests()).isEmpty();
+      checkAppearsActive(tester, true);
+    }, variant: toggleInteractionModeVariant);
+
+    testWidgets('request has an error', (tester) async {
+      final toggleInteractionMode = toggleInteractionModeVariant.currentValue!;
+
+      await setupPage(tester, pageUserId: eg.selfUser.userId);
+      check(store.userSettings.presenceEnabled).isTrue();
+      checkAppearsActive(tester, false);
+
+      // The appearance changes and the request is sent, immediately.
+      prepareRequestError(Duration(milliseconds: 100));
+      await doToggle(tester, toggleInteractionMode);
+      await tester.pump();
+      await tester.pump();
+      checkAppearsActive(tester, true);
+      checkRequest(true);
+
+      // After [PerAccountSettingBuilder.localEchoMinimum], changes back
+      await tester.pump(RemoteSettingBuilder.localEchoMinimum);
+      await tester.pump();
+      checkAppearsActive(tester, false);
+
+      // Wait a while, idly: no change, no extra requests
+      await tester.pump(RemoteSettingBuilder.localEchoIdleTimeout * 2);
+      check(connection.takeRequests()).isEmpty();
+      checkAppearsActive(tester, false);
+    }, variant: toggleInteractionModeVariant);
+
+    testWidgets('spam-tapping', (tester) async {
+      final toggleInteractionMode = toggleInteractionModeVariant.currentValue!;
+
+      await setupPage(tester, pageUserId: eg.selfUser.userId);
+      check(store.userSettings.presenceEnabled).isTrue();
+      checkAppearsActive(tester, false);
+
+      Future<void> doSpamTap() async {
+        final newValue = !getValue(tester);
+        // The appearance changes and the request is sent, immediately.
+        prepareRequestSuccess(Duration(milliseconds: 100));
+        scheduleEventAfter(Duration(milliseconds: 150), newValue);
+        await doToggle(tester, toggleInteractionMode);
+        await tester.pump();
+        await tester.pump();
+        checkAppearsActive(tester, newValue);
+        checkRequest(newValue);
+      }
+
+      // Events will be coming in, but those don't control the switch;
+      // only the user interaction does, until there have been no interactions
+      // for [PerAccountSettingBuilder.localEchoMinimum].
+      await doSpamTap();
+      await tester.pump(Duration(milliseconds: 90));
+      await doSpamTap();
```

Comment:
It doesn't seem like these actually check that events don't control the switch.

In fact I tried deleting the key part of `stop`:
```diff
   Future<void> stop() async {
-    if (_lowerBoundCompleter != null) {
-      await _lowerBoundCompleter!.future;
-      if (_disposed) return;
-    }
     value = OptionNone();
   }
```

and it seems tests still pass.